Foundation/URLSession/http/HTTPURLProtocol.swift

```diff
@@ -12,85 +12,45 @@ import Dispatch
 
 internal class _HTTPURLProtocol: _NativeProtocol {
 
-    fileprivate var easyHandle: _EasyHandle!
-    fileprivate lazy var tempFileURL: URL = {
-        let fileName = NSTemporaryDirectory() + NSUUID().uuidString + ".tmp"
-        _ = FileManager.default.createFile(atPath: fileName, contents: nil)
-        return URL(fileURLWithPath: fileName)
-    }()
-
     public required init(task: URLSessionTask, cachedResponse: CachedURLResponse?, client: URLProtocolClient?) {
-        self.internalState = _InternalState.initial
-        super.init(request: task.originalRequest!, cachedResponse: cachedResponse, client: client)
-        self.task = task
-        self.easyHandle = _EasyHandle(delegate: self)
+        super.init(task: task, cachedResponse: cachedResponse, client: client)
     }
 
     public required init(request: URLRequest, cachedResponse: CachedURLResponse?, client: URLProtocolClient?) {
-        self.internalState = _InternalState.initial
         super.init(request: request, cachedResponse: cachedResponse, client: client)
-        self.easyHandle = _EasyHandle(delegate: self)
     }
 
     override class func canInit(with request: URLRequest) -> Bool {
         guard request.url?.scheme == "http" || request.url?.scheme == "https" else { return false }
         return true
     }
 
-    override class func canonicalRequest(for request: URLRequest) -> URLRequest {
-        return request
-    }
-
-    override func startLoading() {
-        resume()
-    }
-
-    override func stopLoading() {
-        if task?.state == .suspended {
-            suspend()
-        } else {
-            self.internalState = .transferFailed
-            guard let error = self.task?.error else { fatalError() }
-            completeTask(withError: error)
+    override func didReceive(headerData data: Data, contentLength: Int64) -> _EasyHandle._Action {
+        guard case .transferInProgress(let ts) = internalState else {
+            fatalError("Received header data, but no transfer in progress.")
         }
-    }
-
-    /// The internal state that the task is in.
-    ///
-    /// Setting this value will also add / remove the easy handle.
-    /// It is independt of the `state: URLSessionTask.State`. The
-    /// `internalState` tracks the state of transfers / waiting for callbacks.
-    /// The `state` tracks the overall state of the task (running vs.
-    /// completed).
-    fileprivate var internalState: _InternalState {
```

Comment:
The property InternalState has been moved to  NativeProtocol.The git diff is not showing and teh code followed by the property is part of ```didReceve(headerData```.